pkg/skaffold/instrumentation/export.go

```diff
@@ -283,3 +299,160 @@ func errorMetrics(ctx context.Context, meter skaffoldMeter, m metric.Meter, labe
 		unknownCounter.Record(ctx, 1, labels...)
 	}
 }
+
+type TraceExporterConfig struct {
+	writer    io.Writer
+	keyFS     http.FileSystem
+	keyPath   string
+	jaegerURL string
+}
+
+type TraceExporterOption func(te *TraceExporterConfig)
+
+func WithWriter(w io.Writer) TraceExporterOption {
+	return func(teconf *TraceExporterConfig) {
+		teconf.writer = w
+	}
+}
+
+func WithKeyFS(keyFS http.FileSystem) TraceExporterOption {
+	return func(teconf *TraceExporterConfig) {
+		teconf.keyFS = keyFS
+	}
+}
+
+func WithJaegerURL(jaegerURL string) TraceExporterOption {
+	return func(teconf *TraceExporterConfig) {
+		teconf.jaegerURL = jaegerURL
+	}
+}
+
+func WithKeyPath(keyPath string) TraceExporterOption {
+	return func(teconf *TraceExporterConfig) {
+		teconf.keyPath = keyPath
+	}
+}
+
+func initTraceExporter(opts ...TraceExporterOption) (trace.TracerProvider, func(context.Context) error, error) {
+	teconf := TraceExporterConfig{
+		writer:    os.Stdout,
+		keyFS:     nil,
+		keyPath:   "/secret/keys.json",
+		jaegerURL: "http://localhost:14268/api/traces",
```

Comment:
Can this be pushed into the `jaeger` section below?

Reply:
I have removed jaegerURL from the config here and hardcoded it directly into `initJaegerTraceExporter`